sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/PageBlobAsyncClient.java

```diff
@@ -397,19 +407,27 @@
      * @return A reactive response emitting all the different page ranges.
      * @throws IllegalArgumentException If {@code prevSnapshot} is {@code null}
      */
-    public Flux<PageRange> getPageRangesDiff(BlobRange blobRange, String prevSnapshot, BlobAccessConditions accessConditions) {
-        blobRange = blobRange == null ? new BlobRange(0) : blobRange;
-        accessConditions = accessConditions == null ? new BlobAccessConditions() : accessConditions;
+    public PagedFlux<PageRange> getPageRangesDiff(BlobRange blobRange, String prevSnapshot, BlobAccessConditions accessConditions) {
+        final BlobRange finalBlobRange = blobRange == null ? new BlobRange(0) : blobRange;
```

Comment:
Is it necessary to have final on local variable? It seems the logic here is not too complicated to fall into trouble somehow

Reply:
It is not necessary to mark it final. However, the reason the value is assigned to a new variable instead of the old one is because of Java's requirements of what variables you can reference from within lambdas. The compile error that appears is along the lines of "Variables references in lambdas must be final or effectively final." While the reassign makes the variable seen as "effectively final," declaring it final makes it clear that this variable needs to be this way.

Reply:
Reverting this change. No longer necessary while addressing other readability comments.